src/PaymentMethodCollectElement.res

```diff
@@ -30,38 +31,54 @@ let make = (~integrateError, ~logger) => {
 
   // Form a list of available payment methods
   React.useEffect(() => {
-    let availablePMT = {
-      card: [],
-      bankTransfer: [],
-      wallet: [],
-    }
+    let availablePM: array<paymentMethod> = []
+    let availablePMT: array<paymentMethodType> = []
     let _ = options.enabledPaymentMethods->Array.map(pm => {
       switch pm {
-      | Card(cardType) =>
-        if !(availablePMT.card->Array.includes(cardType)) {
-          availablePMT.card->Array.push(cardType)
+      | Card(_) =>
+        if !(availablePM->Array.includes(Card)) {
+          availablePMT->Array.push(Card(Debit))
+          availablePM->Array.push(Card)
+        }
+      | BankTransfer(_) =>
+        if !(availablePM->Array.includes(BankTransfer)) {
+          availablePM->Array.push(BankTransfer)
+        }
+        if !(availablePMT->Array.includes(pm)) {
+          availablePMT->Array.push(pm)
+        }
+      | Wallet(_) =>
+        if !(availablePM->Array.includes(Wallet)) {
+          availablePM->Array.push(Wallet)
         }
-      | BankTransfer(bankTransferType) =>
-        if !(availablePMT.bankTransfer->Array.includes(bankTransferType)) {
-          availablePMT.bankTransfer->Array.push(bankTransferType)
+        if !(availablePMT->Array.includes(pm)) {
+          availablePMT->Array.push(pm)
         }
-      | Wallet(walletType) =>
-        if !(availablePMT.wallet->Array.includes(walletType)) {
-          availablePMT.wallet->Array.push(walletType)
+        if !(availablePMT->Array.includes(pm)) {
+          availablePMT->Array.push(pm)
         }
       }
     })
 
-    let availablePM: array<paymentMethod> = []
-    if !(availablePM->Array.includes(BankTransfer)) && availablePMT.bankTransfer->Array.length > 0 {
-      availablePM->Array.push(BankTransfer)
-    }
-    if !(availablePM->Array.includes(Card)) && availablePMT.card->Array.length > 0 {
-      availablePM->Array.push(Card)
-    }
-    if !(availablePM->Array.includes(Wallet)) && availablePMT.wallet->Array.length > 0 {
-      availablePM->Array.push(Wallet)
-    }
+    // Sort available PM (cards first)
+    availablePM->Array.sort((a, b) =>
```

Comment:
We take an SDK parameter from the merchant called `paymentMethodOrder` where merchant gets to decide the ordering of the Payment Methods. Is that support not required for this widget?

Reply:
Merchants are not integrating this at the moment, but it makes sense to pass this as an input to the SDK rather than writing custom logic in the widget 👍